packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -319,15 +332,22 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
       if (_mode == _RefreshIndicatorMode.drag || _mode == _RefreshIndicatorMode.armed)
         _dismiss(_RefreshIndicatorMode.canceled);
     } else if (notification is ScrollUpdateNotification) {
-      if (_mode == _RefreshIndicatorMode.drag || _mode == _RefreshIndicatorMode.armed) {
+      if (notification.dragDetails != null && (_mode == _RefreshIndicatorMode.drag || _mode == _RefreshIndicatorMode.armed)) {
```

Comment:
Without `notification.dragDetails != null` the test `Refresh starts while scroll view moves back to 0.0 after overscroll` failed, because the indicator kept without a drag gesture moving eventually was cancelled. Previously this didn't happen because of claping on this line

https://github.com/flutter/flutter/blob/6b51924de3575d889802baeb609e1b346c53c297/packages/flutter/lib/src/material/refresh_indicator.dart#L398